sematic/resolvers/cloud_resolver.py

```diff
@@ -231,6 +261,39 @@ def _end_inline_execution(self, future_id):
         """Callback called at the end of an inline execution"""
         logger.info(END_INLINE_RUN_INDICATOR.format(future_id))
 
+    def _can_schedule_future(self, _: AbstractFuture) -> bool:
+        """Return whether we can schedule an external run."""
+        return (
+            not self._max_parallelism or self._remote_runs_count < self._max_parallelism
+        )
+
+    def _inc_remote_runs_count(self, count: int = 1) -> int:
```

Comment:
Will we ever need to increment/decrement by more than 1?

For simplicity's sake, can we do away with these two methods and simply do
```python
self._remote_run_count += 1
```
at the call sites?

The validation seems unnecessary since we have `_can_schedule_future`.

Reply:
The method went away completely.